miri-script/src/commands.rs

```diff
@@ -181,32 +190,32 @@ impl<'a> MiriRunner<'a> {
             return Ok(());
         }
         // We need to build a sysroot.
-        let miri_test_target = std::env::var("MIRI_TEST_TARGET").unwrap_or_default();
-        let target = (!miri_test_target.is_empty()).then_some(miri_test_target.as_str());
+        let miri_test_target = std::env::var_os("MIRI_TEST_TARGET").unwrap_or_default();
+        let target = (!miri_test_target.is_empty()).then_some(miri_test_target.as_os_str());
         let sysroot = self.build_sysroot(target)?;
         self.set_env("MIRI_SYSROOT", sysroot);
         Ok(())
     }
 
     /// Build a sysroot and set MIRI_SYSROOT to use it. Arguments are passed to `cargo miri setup`.
-    fn build_sysroot(&self, target: Option<&str>) -> Result<String> {
+    fn build_sysroot(&self, target: Option<&OsStr>) -> Result<String> {
         let manifest_path = path!(self.miri_dir / "cargo-miri" / "Cargo.toml");
         let manifest_path = manifest_path.to_string_lossy().into_owned();
         let active_toolchain = self.get_toolchain_override();
         let mut args = [
-            &active_toolchain,
-            &self.cargo_extra_flags,
-            "--manifest-path",
-            &manifest_path,
-            "--",
-            "miri",
-            "setup",
+            active_toolchain.as_os_str(),
+            &self.cargo_extra_flags.as_ref(),
+            "--manifest-path".as_ref(),
+            manifest_path.as_ref(),
+            "--".as_ref(),
+            "miri".as_ref(),
+            "setup".as_ref(),
```

Comment:
What an eyesore. :/ We need to find a way to make this not look terrible (as this happens all over the codebase)... isn't duct supposed to be useful for exactly things like this?

Reply:
Yeah, we can probably use `duct` macro here, though it doesn't help with parts where we add some of the arguments conditionally.

Reply:
Ugh. There are two ways to create a command with duct; wiith `cmd` [macro](https://docs.rs/duct/latest/duct/macro.cmd.html) and `cmd` [function](https://docs.rs/duct/latest/duct/fn.cmd.html). The former takes a fixed-size list of arguments that implement `Into<OsString>`, whereas the latter takes an iterable. Neither is ideal: the former takes literals/variable names - so if any of the options is appended conditionally, it won't work - whereas the latter is more flexible and more noisy at the same time.

I honestly think we might as well roll back to std processes.

Reply:
https://github.com/matklad/xshell This lib might be worth a shot as well.